repos/system_upgrade/common/libraries/distro.py

```diff
@@ -16,3 +18,56 @@ def get_distribution_data(distribution):
         raise StopActorExecutionError(
             'Cannot find distribution signature configuration.',
             details={'Problem': 'Distribution {} was not found in {}.'.format(distribution, distributions_path)})
+
+
+def get_distro_repofiles(distro=None):
+    """
+    Get distribution provided repofiles
+
+    Note that this does not perform any validation, the caller must check
+    whether the files exist.
+
+    :return: A list of paths to repofiles provided by distribution
+    :rtype: List(String)
+    """
+    distro = distro or get_distro_id()
+    distro_provided_repofiles = {
+        "rhel": [
+            "/etc/yum.repos.d/redhat.repo",
+        ],
+        "centos": [
+            "/etc/yum.repos.d/centos.repo",
+            "/etc/yum.repos.d/centos-addons.repo",
+        ],
+    }
+    return distro_provided_repofiles[distro]
+
+
+def get_distro_repoids(context, distro=None):
+    """
+    Get repoids defined in distro provided repofiles
+
+    On RHEL this delegates to rhsm.get_available_repo_ids.
+
+    :param context: An instance of mounting.IsolatedActions class
+    :param distro: The distro, defaults to current
+    :return: Repoids of distribution provided repositories
+    :type: List(String)
+    """
+
+    distro = distro or get_distro_id()
+    if distro == 'rhel':
+        if rhsm.skip_rhsm():
+            return []
+        # I kept this todo here
+        # Get the RHSM repos available in the target RHEL container
+        # TODO: very similar thing should happens for all other repofiles in container
+        return rhsm.get_available_repo_ids(context)
+
+    repofiles = repofileutils.get_parsed_repofiles(context)
+    distro_repoids = []
+    for rfile in repofiles:
+        if rfile.file in get_distro_repofiles(distro) and rfile.data:
+            distro_repoids.extend([repo.repoid for repo in rfile.data])
```

Comment:
i am thinking that we shuld actually check whether such repofiles are present on in the given context. what if the installed container does not contain installed a package with required repofiles? it would be nice to have logged whether the file (or some of files) are actually missing - for the debugging purposes.

Reply:
I think we synced about this outside of GitHub without any real conclusion, but I think it would be beneficial to have at least the logs.